docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/hemoglobin/index.rst

```diff
@@ -6,4 +6,237 @@ Hemoglobin Model
+
+.. code-block:: R
+
+  # define constants
+  XMAX = 220
+  EULERS_CONSTANT = 0.57721566490153286060651209008240243104215933593992
+  w = c(0.4,0.6)
+
+  # import standard R functions for the gamma distributions (pgamma and rgamma)
+  pacman::p_load(data.table,actuar)
+
+  # function to calculate gamma distribution parameters from mean and variance
+  gamma_mv2p = function(mn, vr){
+    list(shape = mn^2/vr,rate = mn/vr)}
+
+  # function to calculate mirror gumbel distribution parameters from mean and variance 
+  mgumbel_mv2p = function(mn, vr){
+    list(
+      alpha = XMAX - mn - EULERS_CONSTANT*sqrt(vr)*sqrt(6)/pi,
+      scale = sqrt(vr)*sqrt(6)/pi)}
+  # function to randomly sample n times from mirror gumbel distribution
+  rmgumbel = function(n, alpha, scale){
+    mn = alpha + scale*EULERS_CONSTANT
+    rgumbel(n, alpha+XMAX-(2*mn), scale)}
+
+  # function to sample from ensemble hemoglobin distribution using functions defined above
+    # n = number of samples to take
+    # mn = mean hemoglobin concentration
+    # vr = hemoglobin distribution variance (standard deviation squared)
+    # w = list of ensemble distribution weights c(gamma_weight, mirror_gumbel_weight)
+  sample_from_hemoglobin_distribution <- function(n, mn, vr, w){
+
+    ##parameters
+    params_gamma = gamma_mv2p(mn, vr)
+    params_mgumbel = mgumbel_mv2p(mn, vr)
+
+    ##weighting
+    sample_list = sum(
+      w[1] * rgamma(n, params_gamma$shape, params_gamma$rate), 
+      w[2] * rmgumbel(n, params_mgumbel$alpha, params_mgumbel$scale)
+    )
+    sample_list
+  }
+
+  # function to calculate area under curve below threshold q for mirror gumbel distribution
+  pmgumbel = function(q, alpha, scale, lower.tail){ 
+    #NOTE: with mirroring, take the other tail
+    pgumbel(XMAX-q, alpha, scale, lower.tail=ifelse(lower.tail,FALSE,TRUE))}
+
+  # function to calculate area under curve of hemoglobin ensemble distribution using the functions defined above
+    # q = hemoglobin threshold
+    # mn = mean hemoglobin concentration
+    # vr = hemoglobin variance (standard deviation squared)
+    # w = list of ensemble distribution weights c(gamma_weight, mirror_gumbel_weight)
+  ens_mv2prev <- function(q, mn, vr, w){
+    x = q
+
+    ##parameters
+    params_gamma = gamma_mv2p(mn, vr)
+    params_mgumbel = mgumbel_mv2p(mn, vr)
+
+    ##weighting
+    prev = sum(
+      w[1] * pgamma(x, params_gamma$shape, params_gamma$rate), 
+      w[2] * pmgumbel(x, params_mgumbel$alpha, params_mgumbel$scale, lower.tail=T))
+    prev}
+
+Pregnancy Adjustment
+^^^^^^^^^^^^^^^^^^^^
+
+To sample hemoglobin values for pregnant/lactating women, use the same functions as above, but multiply the hemoglobin mean and standard deviation parameters used for those functions by the respective pregnancy adjustment factors listed below. Notably, the GBD 2019 assumes that the pregnancy adjustment factor applies to 40 weeks of gestation and 6 weeks post-gestation.
```

Comment:
Are the adjustment factors the ratios of pregnant women to all women, or pregnant women to non-pregnant women? This sentence makes it sound like the former, but when you first describe the ratios in the GBD section, it sounded like the latter.

Reply:
I added a note clarifying this in the documentation. My understanding is that the adjustment factor is derived from nonpregnant versus pregnant women, but then is used in GBD to compare pregnant women with the general population.